python-package/lightgbm/basic.py

```diff
@@ -2250,6 +2251,10 @@ def get_data(self):
                     self.data = self.data.iloc[self.used_indices].copy()
                 elif isinstance(self.data, dt_DataTable):
                     self.data = self.data[self.used_indices, :]
+                elif isinstance(self.data, Sequence):
+                    self.data = np.array([self.data[i] for i in self.used_indices])
```

Comment:
I thought we require from `Sequence` implementations to be able pass list as index.
https://github.com/microsoft/LightGBM/blob/5b7a6f3e7150aeb704d1dd2b852d246af3e913a3/python-package/lightgbm/basic.py#L636
https://github.com/microsoft/LightGBM/blob/5b7a6f3e7150aeb704d1dd2b852d246af3e913a3/python-package/lightgbm/basic.py#L645
So maybe we can simplify to `self.data[self.used_indices]`?

Reply:
I've made the change as suggested. The requirement is mentioned as a comment in `Sequence.__getitem__` docstring.

https://github.com/cyfdecyf/LightGBM/blob/sequence-dataset/python-package/lightgbm/basic.py#L648